src/applications/facility-locator/components/LocationHours.jsx

```diff
@@ -63,12 +65,22 @@ export default class LocationHours extends Component {
       if (hours[k] === '-') {
         mappedHours[k] = 'Closed';
       } else if (hours[k]) {
-        const regex = /(\d+\w\w)-(\d+\w\w)/;
+        const regex = /(([1-8][0-9]{2}|9[0-8][0-9]|99[0-9]|1[01][0-9]{2}|12[0-4][0-9]|125[0-9])+\w\w)-(([1-8][0-9]{2}|9[0-8][0-9]|99[0-9]|1[01][0-9]{2}|12[0-4][0-9]|125[0-9])+\w\w)/;
```

Comment:
This is pretty rough to understand. Also, the parsing in this component is just pretty brittle in general.

I think we could get to a simpler, more reliable place by using `moment`. You could split the date string at `-` and then parse each piece using `moment(timeString, "hmma")`. You can then run `isValid()` on the result to see if it parsed correctly. And instead of using the `colonizeTime` function, you can format it like `momentDate.format("h:mma")`.

Reply:
@jbalboni Yeah I didn't really like how this was written. I will do some refactoring so it's better to read and more reliable.